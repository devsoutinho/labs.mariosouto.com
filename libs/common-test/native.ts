import type { Config } from "@jest/types";

const config: Config.InitialOptions = {
  preset: 'jest-expo',
  setupFilesAfterEnv: ['@testing-library/jest-native/extend-expect'],
  clearMocks: true,
};

export default config;
