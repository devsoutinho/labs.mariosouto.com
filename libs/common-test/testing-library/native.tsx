import { render as rtlRender } from '@testing-library/react-native';
export * from '@testing-library/react-native';

interface DefaultWrapperProps {
  children: React.ReactNode;
}
function DefaultWrapper({ children }: DefaultWrapperProps) {
  return children;
}

export const customRender = (AllTheProviders = DefaultWrapper) => {
  return (ui: any, options = {}) => {
    return rtlRender(ui, {
      wrapper: AllTheProviders as any,
      ...options,
    });
  }
}

export const render = () => console.error("Please create your own render method using customRender()");
