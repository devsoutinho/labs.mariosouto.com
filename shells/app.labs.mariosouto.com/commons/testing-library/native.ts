import { render as rtlRender } from "@testing-library/react-native";
export * from "@testing-library/react-native";

interface DefaultWrapperProps {
  children: React.ReactNode;
}
function DefaultWrapper({ children }: DefaultWrapperProps) {
  return children;
}

export const customRender = (AllTheProviders = DefaultWrapper) => {
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  return (ui: any, options = {}) => {
    return rtlRender(ui, {
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      wrapper: AllTheProviders as any,
      ...options,
    });
  };
};

export const render = () =>
  console.error("Please create your own render method using customRender()");
