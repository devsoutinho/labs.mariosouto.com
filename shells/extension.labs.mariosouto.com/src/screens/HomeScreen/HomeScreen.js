import React from "react";
import { Box, Text } from "@devsoutinho/sknui";
export function HomeScreen() {
  return React.createElement(
    Box,
    {
      styleSheet: {
        flex: 1,
        backgroundColor: "#000",
        alignItems: "center",
        justifyContent: "center",
      },
    },
    React.createElement(
      Text,
      {
        styleSheet: {
          color: "#fff",
        },
      },
      "Ol\u00E1 pessoas!"
    )
  );
}
