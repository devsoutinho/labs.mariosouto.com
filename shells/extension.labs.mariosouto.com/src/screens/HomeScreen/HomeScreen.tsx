import React from "react";
import { Box, Text } from "@devsoutinho/sknui/web";

export function HomeScreen() {
  const [message, setMessage] = React.useState("");
  React.useEffect(() => {
    chrome.storage.local.get(["wordCountApp"], (result) => {
      const { length, totalWords } = result.wordCountApp || {};
      setMessage(
        `Selected text length: ${length} | Selected text total words: ${totalWords}`
      );
    });
  }, []);

  return (
    <Box
      styleSheet={{
        width: "320px",
        flex: 1,
        backgroundColor: "#000",
        alignItems: "center",
        justifyContent: "center",
      }}
    >
      <Text
        styleSheet={{
          color: "#fff",
        }}
      >
        {message}
      </Text>
    </Box>
  );
}
