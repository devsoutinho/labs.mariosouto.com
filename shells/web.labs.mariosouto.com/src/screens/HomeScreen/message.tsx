export const message = "OlÃ¡ pessoas!";
