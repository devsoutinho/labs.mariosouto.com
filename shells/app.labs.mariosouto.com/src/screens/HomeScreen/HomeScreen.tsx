import { Box, Text } from "@devsoutinho/sknui/native";

export function HomeScreen() {
return (
<Box styleSheet={{
flex: 1,
backgroundColor: '#000',
alignItems: 'center',
justifyContent: 'center',
}}>
<Text styleSheet={{
color: '#fff',
}}>
Ol√° pessoas!
</Text>
</Box>
);
}