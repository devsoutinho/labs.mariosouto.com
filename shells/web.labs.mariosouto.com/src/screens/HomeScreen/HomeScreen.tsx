import React from "react";
import { Box, Text, Form, InputText, Button, Icon } from "@devsoutinho/sknui/web";
import { message } from "@src/screens/HomeScreen/message";

export function HomeScreen() {
  return (
    <Box
      styleSheet={{
        flex: 1,
        alignItems: "center",
        justifyContent: "center",
      }}
    >
      <Form onSubmit={() => {
        window.location.href = "/dashboard";
      }}>
        <InputText label="Email"  />

        <Button
          type="submit"
          suffix={<Icon name="arrow_left" />}
        >
          Entrar
        </Button>
        <Box
          styleSheet={{
            marginTop: "16px",
          }}
        >
          <Text>
            Ou fa√ßa login com
          </Text>
          <Box>
            <Button>
              Github
            </Button>
          </Box>
        </Box>
      </Form>
    </Box>
  );
}
