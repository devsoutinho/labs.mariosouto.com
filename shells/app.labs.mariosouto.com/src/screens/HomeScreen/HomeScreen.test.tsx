import React from "react";
import { HomeScreen } from "./HomeScreen";
import { customRender } from "@devsoutinho/common-test/testing-library/native";

const render = customRender();

describe("<HomeScreen />", () => {
  it("renders as expected", () => {
    const { toJSON } = render(
      // eslint-disable-next-line @typescript-eslint/no-explicit-any
      <HomeScreen navigation={{} as any} route={{} as any} key={""} />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
