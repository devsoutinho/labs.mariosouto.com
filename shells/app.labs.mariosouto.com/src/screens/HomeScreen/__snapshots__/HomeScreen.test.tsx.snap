// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> renders as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
      }
    }
  >
    Ol√° pessoas!
  </Text>
  <View
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
        }
      }
    >
      Go to Storybook
    </Text>
  </View>
</View>
`;
